src/modules/utils/directory_win.rs

```diff
@@ -115,3 +121,11 @@ pub fn is_write_allowed(folder_path: &str) -> std::result::Result<bool, &'static
 
     Ok(result != 0)
 }
+
+extern "system" {
```

Comment:
```suggestion
#[link(name = "Shlwapi")]
extern "system" {
```

Please add this to ensure `Shlwapi` is linked.